File: app/src/at-trivo/java/asiantech/internship/summer/view_pager/timeline_view_pager/FavoriteTimelineFragment.java
package asiantech.internship.summer.view_pager.timeline_view_pager;

import asiantech.internship.summer.recycler_view.TimeLineFragment;
import asiantech.internship.summer.recycler_view.model.Timeline;

public class FavoriteTimelineFragment extends TimeLineFragment {

    @Override
    protected void setUpDataSet() {
        //set no data
    }

    @Override
    protected void setUpRecyclerViewOnScroll() {
        //set no scroll action
    }

    @Override
    protected void setUpSwipeRefresh() {
        //set no swipe refresh action
        mSwipeRefreshLayout.setEnabled(false);
    }

    public Timeline removeTimelineAt(int position) {
        Timeline timeline = mDataSet.remove(position);
        mAdapter.notifyItemRemoved(position);
        return timeline;
    }

    public void addTimeline(Timeline timeline) {
        mDataSet.add(0, timeline);
        mAdapter.notifyDataSetChanged();
    }

    public void removeTimeline(Timeline timeline) {
        int position = mDataSet.indexOf(timeline);
        if (position >= 0) {
            mDataSet.remove(timeline);
            mAdapter.notifyItemRemoved(position);
        }
    }
}

File: app/src/at-trivo/java/asiantech/internship/summer/viewpagerkotlin/FavoriteTimelineFragment.kt
package asiantech.internship.summer.viewpagerkotlin

import asiantech.internship.summer.recyclerviewkotlin.TimelineFragment
import asiantech.internship.summer.recyclerviewkotlin.model.Timeline

class FavoriteTimelineFragment : TimelineFragment() {
    override fun setUpRecyclerViewAction() {
        swipeRefreshLayout.isEnabled = false
    }

    override fun setUpData() {
    }

    fun removeTimelineAt(position: Int): Timeline? {
        val timeline = dataset.removeAt(position)
        recyclerView.adapter.notifyDataSetChanged()
        return timeline
    }

    fun addTimeline(timeline: Timeline) {
        dataset.add(0, timeline)
        recyclerView.adapter.notifyDataSetChanged()
    }

    fun removeTimeline(timeline: Timeline) {
        val position = dataset.indexOf(timeline)
        if (position >= 0) {
            dataset.remove(timeline)
            recyclerView.adapter.notifyDataSetChanged()
        }
    }

}